src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -41,51 +41,106 @@ const EditBootableVolumesModal: FC<EditBootableVolumesModalProps> = ({
+  const [size, setSize] = useState<string>(initialParams.size);
+  const [sizeOptions, setSizeOptions] = useState<string[]>(
+    instanceTypesAndSizesObject[instanceType] || [],
+  );
+  const [description, setDescription] = useState<string>(initialParams.description);
+
+  const onInstanceTypeChange = (value: string) => {
+    setInstanceType(value);
+    setSizeOptions(instanceTypesAndSizesObject[value]);
+    setSize(instanceTypesAndSizesObject[value][0]);
+  };
+
+  const onChangeVolumeParams = () => {
+    const preferenceLabel = preference && { [DEFAULT_PREFERENCE_LABEL]: preference };
+    const instanceLabel = instanceType && {
+      [DEFAULT_INSTANCETYPE_LABEL]: `${instanceType}.${size}`,
+    };
+    const descriptionAnnotation = description && { [ANNOTATIONS.description]: description };
+
+    const metadata: BootableVolumeMetadata = {
+      labels: {
+        ...dataSource?.metadata?.labels,
+        ...preferenceLabel,
+        ...instanceLabel,
+      },
+      annotations: {
+        ...dataSource?.metadata?.annotations,
+        ...descriptionAnnotation,
+      },
+    };
+
+    return changeBootableVolumeMetadata(dataSource, metadata);
+  };
 
   return (
     <TabModal<K8sResourceCommon>
       obj={dataSource}
       isOpen={isOpen}
       onClose={onClose}
       headerText={t('Edit')}
-      onSubmit={changeBootableVolumeMetadata(dataSource, initialMetadata, metadata)}
+      onSubmit={onChangeVolumeParams()}
     >
       <Form>
         <FormGroup label={t('Default preference')} isRequired>
           <FilterSelect
-            selected={metadata?.labels?.[DEFAULT_PREFERENCE_LABEL]}
-            setSelected={setBootableVolumeMetadata('labels', DEFAULT_PREFERENCE_LABEL)}
+            selected={preference}
+            setSelected={setPreference}
             options={preferencesNames}
             groupVersionKind={VirtualMachineClusterPreferenceModelGroupVersionKind}
             optionLabelText={t('preference')}
           />
         </FormGroup>
-        <FormGroup label={t('Default Instancetype')}>
-          <FilterSelect
-            selected={metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]}
-            setSelected={setBootableVolumeMetadata('labels', DEFAULT_INSTANCETYPE_LABEL)}
-            options={instanceTypesNames}
-            groupVersionKind={VirtualMachineClusterInstancetypeModelGroupVersionKind}
-            optionLabelText={t('Instancetype')}
-          />
-        </FormGroup>
+        <Grid hasGutter>
+          <GridItem span={6}>
+            <FormGroup label={t('Default Instancetype')}>
+              <FilterSelect
+                selected={instanceType}
+                setSelected={onInstanceTypeChange}
+                options={Object.keys(instanceTypesAndSizesObject) || []}
+                groupVersionKind={VirtualMachineClusterInstancetypeModelGroupVersionKind}
+                optionLabelText={t('Instancetype')}
```

Comment:
I think this should be 'InstanceType' based on what you told me today.

Reply:
Yes, and that is fixed in a different PR: https://github.com/kubevirt-ui/kubevirt-plugin/pull/1056